pkg/catalog/manager/traverse.go

```diff
@@ -311,10 +324,19 @@ type catalogYml struct {
 	Labels     map[string]string `yaml:"labels,omitempty"`
 }
 
-func setCatalogErrorState(cmt *CatalogInfo, catalog *v3.Catalog, projectCatalog *v3.ProjectCatalog, clusterCatalog *v3.ClusterCatalog) {
-	v3.CatalogConditionRefreshed.False(catalog)
-	v3.CatalogConditionRefreshed.Message(catalog, fmt.Sprintf("Error syncing catalog %v", catalog.Name))
-	cmt.catalog = catalog
-	cmt.projectCatalog = projectCatalog
-	cmt.clusterCatalog = clusterCatalog
+func processInvalidChartErrors(entriesWithErrors []ChartsWithErrors) string {
```

Comment:
Use a string builder with the loop, it will be cleaner/faster/less memory https://golang.org/pkg/strings/#Builder